server/routes/api.js

```diff
@@ -369,26 +369,31 @@ router.get('/messages/:id', async (req, res, next) => {
 
       const pdwMode = nconf.get('messages:pdwMode');
 
-      console.time('sql');
-      const queryResult = await Message.query()
-          .findById(req.params.id)
-          .eager('alias(messageView)')
-          .modify('defaultSort')
-          .modify(builder => {
-              if (HideCapcode && !req.isAuthenticated())
-                  builder.omit(Message, ['address']);
-          });
-
-      console.timeEnd('sql');
-      console.time('send');
-
-      if (!queryResult || (queryResult.ignore || (pdwMode && !queryResult.alias)))
-          res.status(200).json({});
-      else
-          res.status(200).json({queryResult});
-
-      queryResult.catch(err => {res.status(500).send(err)});
-      console.timeEnd('send');
+    try {
+        console.time('sql');
+        const queryResult = await Message.query()
+            .findById(req.params.id)
+            .eager('alias(messageView)')
+            .modify('defaultSort')
+            .modify(builder => {
+                if (HideCapcode && !req.isAuthenticated())
+                    builder.omit(Message, ['address']);
+            });
+
+        console.timeEnd('sql');
+        console.time('send');
+
+        if (!queryResult || (queryResult.ignore || (pdwMode && !queryResult.alias)))
+            res.status(200).json({});
```

Comment:
Think we are missing a time end here

Reply:
Nope, comes after the if/else to include empty result handling AND regular result handling.

Reply:
Line 390 if you ask GitHub; 394 if you ask my IDE. I'm kinda worried where the 3 lines are that I'm not aware of ...